src/gc.c

```diff
@@ -735,6 +735,8 @@ JL_DLLEXPORT jl_value_t *jl_gc_big_alloc(jl_ptls_t ptls, size_t sz)
 {
     maybe_collect(ptls);
     size_t offs = offsetof(bigval_t, header);
+    assert(sz > sizeof(jl_taggedvalue_t) && "sz must include tag");
```

Comment:
`sz >= sizeof(jl_taggedvalue_t)` otherwise `MEMDEBUG=1` is going to be unhappy about allocating singletons.